### PR TITLE
(895) Add page to view submitted referral additional info

### PR DESCRIPTION
**Pull request by Gweaton**
## Context

https://trello.com/c/LXCt2U2V/895-add-page-for-viewing-a-referrals-additional-information-s

## Changes in this PR

Adds page for viewing additional information on a submitted referral

### After

image

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes API for this change to work...
  - [ ] ... and they have been released to production already

### Post-merge

- [ ] This adds, extends or meaningfully modifies a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [ ] This makes new expectations of the API...
  - [ ] ... and I have notified the API developer(s) of changes to the contract tests (Pact), or the API is already compliant
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

File: integration_tests/e2eReferDisabled/refer.cy.ts [hunk 1]
```diff
@@ -203,6 +203,20 @@ context('Refer', () => {
     notFoundPage.shouldContain404H2()
   })
 
+  it("Doesn't show the additional information page for a submitted referral", () => {
```
Comment (edavey) [inline integration_tests/e2eReferDisabled/refer.cy.ts:206]: again, would it be helpful to be more explicit that an error page is shown? (If I've understood correctly what's meant by `Page.verifyOnPage(NotFoundPage)`! )

File: integration_tests/pages/refer/show/additionalInformation.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  shouldContainAdditionalInformation() {
+    cy.get('[data-testid="submitted-text"]').should(
+      'contain.text',
+      `Submitted in referral on ${DateUtils.govukFormattedFullDateString(this.referral.submittedOn)}.`,
+    )
+  }
+
+  shouldContainSubmittedText() {
+    cy.get('[data-testid="additional-information"]').should('have.text', this.referral.additionalInformation)
```
Comment (edavey) [inline integration_tests/pages/refer/show/additionalInformation.ts:26]: probably I'm not understand how/where the `referral` test fixture is defined but I can't see here or in the cypress test how we're verifying that `this.referral.additionaInformation` isn't null/empty/undefined...

**Review by edavey — APPROVED** [review]
👍

File: integration_tests/pages/refer/show/additionalInformation.ts (hunk 2 above)
Comment (yndajas) [reply integration_tests/pages/refer/show/additionalInformation.ts:26]: Not entirely related, but it looks like the names of the two `should` methods are the wrong way round here

Comment (Gweaton) [reply integration_tests/pages/refer/show/additionalInformation.ts]: @edavey Good point. We pass a Referral to this page class on instantiation, and it's a required field on a Referral so I think this is ok for now. I've spotted a couple of things in our tests that could probably be made more robust, which we can address in a future task. I'll add a card to our backlog for this.

@yndajas good spot! Must have been my end of the day brain. Have fixed this.

File: integration_tests/e2eReferDisabled/refer.cy.ts (hunk 1 above)
Comment (Gweaton) [reply integration_tests/e2eReferDisabled/refer.cy.ts:208]: As discussed on the other PR, I'll leave this as-is for now as it's going to be removed very soon and is consistent with other tests.

**Pull request merged**